string goalType

float64 source
float64 destination

